packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -944,21 +944,7 @@ export function forceRehydration(vm: VM) {
 }
 
 export function runFormAssociatedCustomElementCallback(vm: VM, faceCb: () => void) {
-    const {
-        renderMode,
-        shadowMode,
-        def: { formAssociated },
-    } = vm;
-
-    // Technically the UpgradableConstructor always sets `static formAssociated = true` but silently fail here to match browser behavior.
-    if (isUndefined(formAssociated) || isFalse(formAssociated)) {
-        if (process.env.NODE_ENV !== 'production') {
-            logWarn(
-                `Form associated lifecycle methods must have the 'static formAssociated' value set in the component's prototype chain.`
-            );
-        }
-        return;
-    }
```

Comment:
This warning was only needed because it was possible to still access face even though `formAssociated` was not set (because we were previously always setting it to true).

Since that is no longer the case, we can remove this warning now.